src/alloc.rs

```diff
@@ -23,6 +23,8 @@ static ALLOC_LOCK: AllocLock = AllocLock::new();
 
 static mut LAST_ALLOCED: *mut BlockHeader = ::std::ptr::null_mut();
 
+pub static mut HEAP_MAX: usize = 0;
```

Comment:
`HEAP_MAX` sounds like it's related to the size of the heap: `HEAP_HIGH` might be clearer?

And then shouldn't we be tracking both `HEAP_LOW` and `HEAP_HIGH` not just the latter?

Reply:
Maybe `HEAP_TOP`? As it's similar to our descriptions of the stack. 

We don't need to track the lower bound of the heap because it always starts at the end of `.bss` and the linker already gives us a symbol we can use to get that address.

Reply:
Works for me.

> We don't need to track the lower bound of the heap because it always starts at the end of .bss

Aha, can we document that somewhere?

Reply:
Sure. Fixed: ab74aea33982c5397306b6a45e83d4991ab5699a